Classes/Views/GLFileView.m

```diff
@@ -268,6 +260,38 @@ - (NSString *) parseBlame:(NSString *)txt
 	return (NSString *)res;
 }
 
+- (NSString *) htmlHistory:(PBGitTree *)file
+{
+	// \0 can't be passed as a shell argument, so use a sufficiently long random seperator instead
```

Comment:
Eww, that's gross 😉. As I was curious, I investigated a bit more, so that would be ❤️ if that comment said "NSTask internally parses its arguments using -fileSystemRepresentation, which chokes on \0, we must […]".

Reply:
Yeah, I would have preferred to switch this over to libgit2 but it [doesn't appear to support `log --follow` yet.](https://github.com/libgit2/libgit2/issues/3041) So instead we get... this, for now.

It's not just that NSTask and -fileSystemRepresentation don't support null bytes, it's that posix program arguments are null-terminated strings. The kernel/execve won't copy past the first null byte, so nothing above tries to handle them either. Would a better wording be "\0 can't be passed in a posix program argument"?